### ISSUE TITLE
TR2X bug: Game doesn't start instantaneous after Load Game or New Game

### ISSUE THREAD

**Issue by aredfan** [LostArtefacts/TR2X#114]
**Dev snapshot:** 0.2-141-g5752c50

____________________

In both videos, the first part is TR2X and the second part is OG.

**Load Game**

The holdup is when the menu/inventory ring closes.

https://github.com/user-attachments/assets/539aa877-547a-41c5-bd9d-87b08235e200

**New Game**

The holdup is a black screen.

https://github.com/user-attachments/assets/d4f65d8c-8ff6-43b0-ba2f-c477a3160d7a

**Comment by Richard-L** [comment]
I believe this might be the same as https://github.com/LostArtefacts/TR1X/issues/1335

**Comment by rr-** [comment]
Actually I think this is the same issue as this: https://github.com/LostArtefacts/TRX/issues/1597

**Comment by rr-** [comment]
@aredfan can you play around with this version?
https://nebula.wind.garden/serve/pub/TR2X-0.2-143-g10d8d69-dirty-Windows.zip
I've updated the audio engine to convert samples on-demand, upon first use. However, since we're still not using background threads for this task, this change could potentially result in occasional audio clicks. To help in assessing how well this works, we log the time each conversion task takes in milliseconds.

**Comment by aredfan** [comment]
@rr- Yep, always happy to help out.

I played through the Opera House level with this build and I believe I didn't encounter any audio clicks. I also checked the TR2X.log after quitting the game a few times - the audio sample conversion is consistently 0 ms across the board.

My PC is coming up to 4 years old if it's relevant.

 I also noticed the holdup issue for New Game and Load Game is fixed in this build.
